services/cognito/app/resources/user_resource.rb

```diff
@@ -6,13 +6,13 @@ class UserResource < Cognito::ApplicationResource
   filter :primary_identifier
 
   filter :query, apply: lambda { |records, value, _options|
-    query_by_id = "users.id IN (#{value[0]})"
-    query_by_non_id_attrs = %w[primary_identifier first_name last_name email_address]
-                            .map { |field| "#{field} ILIKE '%#{value[0]}%'" }
+    query_by_id = 'users.id IN (:id_query)'
```

Comment:
can't we simply do:

records = records.where(query_by_non_id_attrs)
records.or(records.where(id: value[0])) if /\D/.match?(value[0])

Reply:
Done